### PR TITLE
Replace deprecated `evaluateTransactionFee` with `getMinFeeTx`

### PR DESCRIPTION
**Pull request by Anviking**
- [x] Replace deprecated evaluateTransactionFee with getMinFeeTx in a minimal way (keep using our own `sizeOf_BootstrapWitnesses`)

### Issue Number

ADP-3334

### CODE REVIEW

File: lib/balance-tx/cardano-balance-tx.cabal [hunk 1]
```diff
@@ -75,6 +76,7 @@ library internal
     , groups
     , int-cast
     , lens
+    , memory
```
Comment (Anviking) [inline lib/balance-tx/cardano-balance-tx.cabal]:
```suggestion
```

File: lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Sign.hs [hunk 2]
```diff
@@ -418,3 +404,73 @@ estimateMaxWitnessRequiredPerInput = \case
...
+          { Byron.aaVKDerivationPath = Just $ Byron.HDAddressPayload addrPayload
+          , Byron.aaNetworkMagic = Byron.NetworkTestnet maxBound
+          }
+      where
+        -- Largest address payload used by byron or icarus wallets.
+        --
+        -- Generated using:
+        -- >>> CBOR.toStrictByteString $ Cardano.Byron.Codec.Cbor.encodeDerivationPathAttr passphrase maxBound maxBound
+        addrPayload = unsafeFromHex
+            "01581e581cdab6b4557a5499c7265a1512045e6f29a371bda7894d03f2a041e7d3"
```
Comment (Anviking) [inline lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Sign.hs]: This is unused; probably better to remove for now

```suggestion
```

File: lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx.hs [hunk 3]
```diff
@@ -16,10 +15,6 @@
 {-# LANGUAGE UndecidableInstances #-}
 {-# LANGUAGE UndecidableSuperClasses #-}
 
-{-# OPTIONS_GHC -Wno-warnings-deprecations #-}
--- For 'Shelley.evaluateTransactionFee', see ADP-3334
```
Comment (paweljakubas) [inline lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx.hs:20]: :+1:

**Review by paweljakubas — APPROVED** [review]
Adept refactoring :gem: 

I wonder given those quite numerous int cast whether it is not reasonable to refactor upstream the types and use `Word` rather than `Int` etc. It would be even more straightforward to use `getMinFeeTx`

**Pull request merged**